write-fonts/src/pens.rs

```diff
@@ -0,0 +1,369 @@
+//! Pen implementations based on <https://github.com/fonttools/fonttools/tree/main/Lib/fontTools/pens>
+
+use font_types::{Pen, PenCommand};
+use kurbo::{Affine, BezPath, Point};
+
+/// A pen that transforms params using [kurbo::Affine].
+pub struct TransformPen<'a, T: Pen> {
```

Comment:
maybe the `pens` module could be turned in to a directory containing several modules for each pen? It matches the approach taken in fontTools, and gives each pen greater visibility I believe. Currently in here there's three new pens defined in here, TransformPen, BezPathPen and ReverseContourPen. As the number of pens increase this can get crowded

Reply:
I don't have strong feelings about it. I think in Rust the documentation system would split the types out independent of what files they live in so perhaps this sort of organization is less important?

Reply:
I confirmed theory above with Rust people who know more than me so I am inclined to leave this for now

Reply:
yea my personal thinking here is that it makes sense to split things out when a single file would start to get unwieldy, and there is a nice clear way things could be separated.

Reply:
When would you typically feel a single file is unwieldy? Is it lines, number of types, ...?

For me it would probably be as you got into thousands of lines or the file started to contain increasingly unrelated ideas. Concretely, to me this file is ... wieldy.